subsys/debug/Kconfig

```diff
@@ -159,6 +159,14 @@ config FORCE_NO_ASSERT
 	  CFLAGS and not Kconfig.  Added solely to be able to work
 	  around compiler bugs for specific tests.
 
+config CONFIG_POST_ASSERT_ARCH_DEFINED
+	bool "Make Arch define post assert behavior"
+    default y if ARCH_POSIX
+	help
+	  This boolean option enables architecure specific implemenation of
+	  post assertion behavior. Added to let architecture define if and how
```

Comment:
Nit: "if and how to reset when an assert occurs." Implies the arch would want to reset. Maybe you want to say "what to do after an asserts occurs"